src/cobalt.rs

```diff
@@ -155,27 +155,36 @@ pub fn build(config: &Config) -> Result<()> {
                     reader.read_to_string(&mut text)?;
                     data = toml::from_str(&text)?;
                 } else {
+                    trace!("Skipping loading of data {:?}: unknown file type.", full_path);
+                    trace!("Supported data files extensions are: yml, yaml, json and toml.");
                     continue
                 }
 
                 // now find the nested map it is supposed to be in
                 let mut target_map = if let Some(path) = df.parent() {
                     let mut map = &mut data_map;
-                    for part in path.to_str().unwrap().split("/") {
+                    for part in path.iter() {
                         if part.len() == 0 { continue }
-                        let mut cur_map = map;
-                        map = cur_map
-                            .entry(String::from(part))
-                            .or_insert_with(|| Value::Object(HashMap::new()))
-                            .as_object_mut().unwrap();
+                        if let Some(key) = part.to_str() {
+                            let mut cur_map = map;
+                            map = cur_map
+                                .entry(String::from(key))
+                                .or_insert_with(|| Value::Object(HashMap::new()))
+                                .as_object_mut().unwrap();
+                        } else {
+                            trace!("Skipping loading of data in {:?}: can't convert path to utf-8", path);
+                            break
```

Comment:
Won't this cause us to insert the data in a the wrong section because of the break?

Maybe the data loading should be moved into a separate function.  At that point, it can return early from this function, bypassing the insert below.

Reply:
True.

I've made it its own function and also decided to move to explict exection-stopping errors in these cases to force the user to clean up their data..